pkg/build/strategies/sti/sti.go

```diff
@@ -591,16 +590,17 @@ func (builder *STI) Execute(command string, user string, config *api.Config) err
 	go dockerpkg.StreamContainerIO(errReader, &errOutput, func(a ...interface{}) { glog.Info(a...) })
 
 	err := builder.docker.RunContainer(opts)
-	if util.IsTimeoutError(err) {
-		// Cancel waiting for source input if the container timeouts
-		wg.Done()
-	}
 	if e, ok := err.(errors.ContainerError); ok {
 		// even with deferred close above, close errReader now so we avoid data race condition on errOutput;
 		// closing will cause StreamContainerIO to exit, thus releasing the writer in the equation
 		errReader.Close()
 		return errors.NewContainerError(config.BuilderImage, e.ErrorCode, errOutput)
 	}
+	// Do not wait for source input if the container times out.
+	// FIXME: this potentially leaks a goroutine.
+	if !util.IsTimeoutError(err) {
```

Comment:
On Thursday, September 1, 2016, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/build/strategies/sti/sti.go
> <https://github.com/openshift/source-to-image/pull/581#discussion_r77222225>
> :
>
> >  	if e, ok := err.(errors.ContainerError); ok {
> >  		// even with deferred close above, close errReader now so we avoid data race condition on errOutput;
> >  		// closing will cause StreamContainerIO to exit, thus releasing the writer in the equation
> >  		errReader.Close()
> >  		return errors.NewContainerError(config.BuilderImage, e.ErrorCode, errOutput)
> >  	}
> > +	// Do not wait for source input if the container times out.
> > +	// FIXME: this potentially leaks a goroutine.
> > +	if !util.IsTimeoutError(err) {
>
> See https://play.golang.org/p/kpKJh9Prei
>
> If you never call wg.Add, it is perfectly ok and correct to call wg.Wait(),
> so we don't need to add that extra condition, it is irrelevant.
>
Ah - ok yeah that makes sense now - the count is still 0.


> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/source-to-image/pull/581/files/0b63dd20f38c0ab2c07a571708273cdb43d79f9c#r77222225>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbadNxDN11QXZcFyt1lsRHS9kGvIPLYks5qlxGigaJpZM4Jy8x3>
> .
>